### PR TITLE
highlight exotic pluggable perks

### PR DESCRIPTION
**Pull request by delphiactual**
image

### CODE REVIEW

File: src/app/item-popup/ItemSocketsGeneral.tsx [hunk 1]
```diff
@@ -72,9 +72,16 @@ function ItemSocketsGeneral({
     return null;
   }
 
-  const exoticArmorPerk = item.sockets.categories.find(
-    (c) => c.category.hash === SocketCategoryHashes.ArmorPerks_LargePerk
-  )?.sockets[0];
+  const exoticPluggable =
+    item.bucket.inArmor &&
+    item.sockets?.allSockets.find((socket) =>
+      socket.plugged?.plugDef.plug.plugCategoryIdentifier.match(/enhancements\.exotic/i)
```
Comment (delphiactual) [inline src/app/item-popup/ItemSocketsGeneral.tsx:78]: not sure if is should do `.test(/enhancements\.exotic/i)` or `.includes("enhancements.exotic")` instead.

File: src/app/item-popup/ItemSocketsGeneral.tsx [hunk 2]
```diff
@@ -72,9 +72,16 @@ function ItemSocketsGeneral({
     return null;
   }
 
-  const exoticArmorPerk = item.sockets.categories.find(
-    (c) => c.category.hash === SocketCategoryHashes.ArmorPerks_LargePerk
-  )?.sockets[0];
+  const exoticPluggable =
+    item.bucket.inArmor &&
+    item.sockets?.allSockets.find((socket) =>
```
Comment (delphiactual) [inline src/app/item-popup/ItemSocketsGeneral.tsx:77]:
```suggestion
    item.sockets.allSockets.find((socket) =>
```

File: src/app/item-popup/ItemSocketsGeneral.tsx [hunk 3]
```diff
@@ -72,9 +72,14 @@ function ItemSocketsGeneral({
...
-  const exoticArmorPerk = item.sockets.categories.find(
-    (c) => c.category.hash === SocketCategoryHashes.ArmorPerks_LargePerk
-  )?.sockets[0];
+  const exoticArmorPerk =
+    item.sockets.categories.find(
+      (c) => c.category.hash === SocketCategoryHashes.ArmorPerks_LargePerk
+    )?.sockets[0] ||
+    (item.bucket.inArmor &&
+      item.sockets.allSockets.find((socket) =>
+        socket.plugged?.plugDef.plug.plugCategoryIdentifier.match(/enhancements\.exotic/i)
```
Comment (delphiactual) [inline src/app/item-popup/ItemSocketsGeneral.tsx:81]: not sure if I should use .test(/enhancements\.exotic/i) or .includes("enhancements.exotic") instead.

Comment (nev-r) [reply src/app/item-popup/ItemSocketsGeneral.tsx]: i lean strongly toward includes and trusting the capitalization not to change

**Review by nev-r — APPROVED** [review]
haggling over a millisecond here, but by checking if the item is inArmor, _and_ is exotic first,
you can avoid running a find loop and a bunch of string dissections for every single armor piece this runs into

**Pull request merged**